modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -609,9 +609,21 @@ def mask(
         if row_labels is not None:
             row_positions = self.index.get_indexer_for(row_labels)
         if row_positions is not None:
+            sorted_row_positions = (
+                row_positions
+                if (
+                    (is_range_like(row_positions) and row_positions.step > 0)
```

Comment:
I see that there's an [existing comment](https://github.com/modin-project/modin/blob/0ae52603f526bc2efee8e6c131613758712d14fc/modin/core/dataframe/pandas/dataframe/dataframe.py#L629) that "Fast range processing of non-1-step ranges is not yet supported." Why is that?

Reply:
It actually should be "... processing of <i>non-positive</i>-step ranges ...".

The "fast" case of masking is when the indices are monotonically (ascending or decreasing) ordered, this way we can just select required indices without shuffling partitions. Range-like indices always produce monotonic sequences, so their processing always should be fast. However, the current masking algorithm only does fast-path if the indices are [monotonically increasing](https://github.com/modin-project/modin/blob/0ae52603f526bc2efee8e6c131613758712d14fc/modin/core/dataframe/pandas/dataframe/dataframe.py#L708-L723). There's a space for improvement by implementing the fast-path for monotonically decreasing indices:
```
df.mask(row_indices=slice(None, None, -1)) # reverse the frame:

--part1--                            --part1(reversed)--					--part3(reversed)--
1		                       4                                                          13
2                                      3                                                          12
3                                      2                                                          11
4                                      1                                                          10
--part2--                            --part2(reversed)--                                          9
5                                      8                                                        --part2(reversed)--
6               apply indexer          7                         reorder partitions               8
7               to each partition      6                         (no shuffling required)          7
8		================>      5			 =======================>         6
--part3--       ================>    --part3(reversed)--         =======================>         5
9                                      13                                                       --part1(reversed)--
10                                     12							  4
11                                     11                                                         3
12                                     10                                                         2
13                                     9                                                          1
--------                             --------------------                                       ---------------------
```
In the current `.mask()` implementation, the reversing-indexer would trigger full axis concatenation to reorder rows.

Reply:
I have corrected in-code comment:
"non-1-step ranges -> non-positive-step ranges"